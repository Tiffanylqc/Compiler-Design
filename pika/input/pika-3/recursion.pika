func fib<int n>->int {
	if (n == 0 || n == 1) {
		return 1.
	} else {
		return fib(n - 1) + fib(n - 2).
	}
}

exec {
	var n := 1.
	while (n < 20) {
		print "Fib(", n, ") is ", fib(n), _n_.
		n := n + 1.
	}
}