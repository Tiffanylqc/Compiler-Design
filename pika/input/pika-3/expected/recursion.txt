Fib(1) is 1
Fib(2) is 2
Fib(3) is 3
Fib(4) is 5
Fib(5) is 8
Fib(6) is 13
Fib(7) is 21
Fib(8) is 34
Fib(9) is 55
Fib(10) is 89
Fib(11) is 144
Fib(12) is 233
Fib(13) is 377
Fib(14) is 610
Fib(15) is 987
Fib(16) is 1597
Fib(17) is 2584
Fib(18) is 4181
Fib(19) is 6765
